### DISCUSSION TITLE
Copilot keeps "encountering temporary issues"

### DISCUSSION THREAD

**Discussion by NyankoSong** [community/community#130976] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hello!

I'm using Copilot in a remote connection environment within VSCode. My remote server has only an IPv6 address, so I've set up a proxy server to get things working. While Copilot successfully logs in, and Copilot Chat interacts without issues, the inline Copilot completions aren't functioning. It constantly shows "Copilot is encountering temporary issues." There are no apparent error messages in the Output, which is as follows:

```
2024-07-04 16:55:50.469 [info] [fetcher] Using Helix fetcher, Electron fetcher is not available.
2024-07-04 16:55:50.470 [info] [code-referencing] Public code references are enabled.
2024-07-04 16:55:50.564 [info] [auth] Sucessfully authenticated
2024-07-04 16:57:16.663 [info] [fetchCompletions] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex/completions] took 1195 ms
```

I have tried uninstalling and reinstall it, and updating it to the latest version, but it didn't work.

Could anyone help diagnose this issue or suggest a solution? Thank you!

**Comment by Akash1134** [comment, marked as answer]
Hi @NyankoSong ,

Ensure the following settings are properly configured:
- `github.copilot.inlineSuggest.enable`: Set to true.
- `http.proxySupport`: Set to on if you are behind a proxy.
- If you make any changes, try disabling and then re-enabling the GitHub Copilot extension in VSCode.

Also. please add any [URLs to the allowlist](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-firewall-settings-for-github-copilot) as required to setup your proxy connection correctly. 

Thanks!

**Reply by NyankoSong** [reply]
Thanks! I set "http.proxySupport" to "on" (from "override") and everything seems to be working fine. Just out of curiosity, both "on" and "override" indicate that proxy support will be enabled for the extension, so why would the default "override" lead to this issue?

**Reply by Akash1134** [reply]
The difference between "on" and "override"  can sometimes affect how the proxy settings are applied:

For instance, "on" enables proxy support for extensions and uses the proxy settings defined in your environment (e.g., system proxy settings in VSCode), whereas, "override" enables proxy support but allows extensions to **override the proxy settings**, potentially causing conflicts if the extensions specify different proxy settings. In your case, setting it to "on" ensures that the proxy settings are consistently applied across all extensions, avoiding conflicts or misconfigurations that might occur with "override".
